src/DefaultBuilder/test/Microsoft.AspNetCore.NativeAotTests/SlimBuilderDoesNotDependOnX509Test.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Builder;
+
+#nullable enable
+
+var builder = WebApplication.CreateSlimBuilder(args);
+_ = builder.Build();
+
+if (GetType("System.Security.Cryptography", "System.Security.Cryptography.X509Certificates.X509Certificate") is not null) {
```

Comment:
Here's an idea, and you can feel free to tell me to go away:

A test like this could pass because `X509Certificate` has been trimmed away, but it could also pass because of a typo in the type name.

Can we verify that `X509Certificate` is present when the regular `CreateBuilder` is used? Either:
* Have a separate console app that uses `CreateBuilder` and share this assert between them
* Or have a compiler constant + `#ifdef` and run the file twice with different compiler constant values (this would probably require complex build changes).

Reply:
I used `CreateBuilder` as part of my local validation that the test would catch regressions, but it actually fails to build because others parts of `CreateBuilder` aren't AOT compatible. :cry:

Reply:
I wonder if I could write that as a trimming test though...

Reply:
`CreateBuilder `works with AOT. The gRPC template uses it and it builds with AOT without warnings: https://github.com/dotnet/aspnetcore/blob/c5a9c5973bf735f1bb9d611cd50c796f6c32557e/src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Program.cs#L3

Reply:
Alright, then I just need to figure out the combination of feature switches that will make that true in the test.

Reply:
It's still getting trimmed with `CreateBuilder`.  Rather than investigate deeply, I added a test confirming it's preserved when `UseHttps` is invoked.  Now let's confirm I didn't make it depend on having a dev cert...